src/states/elder/mod.rs

```diff
@@ -1064,6 +1065,11 @@ impl Elder {
         if let Some(msg) = self.parsec_map.create_gossip(version, &gossip_target) {
             self.send_direct_message(&gossip_target, msg);
         }
+
+        if self.parsec_map_mut().needs_pruning() {
```

Comment:
not a big issue, however do we need `mut` here?
maybe `self.parsec_map.needs_pruning` will be enough?

Reply:
I don't there is a `parsec_map()` method

Reply:
yeah, but here you have the access to the parsec_map so don't `have to` use that method?

Reply:
Ah yes I see what you mean, yep that's a good point. We do mutate it 2 lines below but I'll update